src/magicgui/type_map/_type_map.py

```diff
@@ -532,3 +539,12 @@ def type2callback(type_: type) -> list[ReturnCallback]:
         if safe_issubclass(type_, registered_type):
             return _RETURN_CALLBACKS[registered_type]
     return []
+
+
+def _is_optional(type_: Any) -> tuple[bool, type]:
```

Comment:
from documentation optional is equivalent of `Union[X, None]` with order. Did you meet in a different order, or did you write this in this way to guarantee safety?

Reply:
it's true that `Optional[X]` results in `Union[X, None]` ... but it's not the only way to create an optional in later python versions

```python
In [2]: get_args(int | None)
Out[2]: (<class 'int'>, <class 'NoneType'>)

In [3]: get_args(None | int)
Out[3]: (<class 'NoneType'>, <class 'int'>)
```